pkg/datastore/mysql/query.go

```diff
@@ -93,3 +98,47 @@ func toMySQLDirection(d datastore.OrderDirection) string {
 		return ""
 	}
 }
+
+func refineOrdersField(orders []datastore.Order) []datastore.Order {
+	for i, order := range orders {
+		switch order.Field {
+		case "SyncState.Status":
+			orders[i].Field = "SyncState"
+		default:
+			continue
+		}
+	}
+	return orders
+}
+
+func refineFiltersField(filters []datastore.ListFilter) []datastore.ListFilter {
+	for i, filter := range filters {
+		switch filter.Field {
+		case "SyncState.Status":
+			filters[i].Field = "SyncState"
+		default:
+			continue
+		}
+	}
+	return filters
+}
+
+func refineFiltersOperator(filters []datastore.ListFilter) ([]datastore.ListFilter, error) {
+	var err error
+	for i, filter := range filters {
+		switch filter.Operator {
+		case "==":
+			filters[i].Operator = "="
+		case "in":
+			filters[i].Operator = "IN"
+		case "not-in":
+			filters[i].Operator = "NOT IN"
+		case "!=", ">", ">=", "<", "<=":
+			continue
```

Comment:
Why not adding?

Reply:
I modify the given `filter` itself not creating the new one, so instead of `adding`, those operators are all supported operators, that why we should just leave it as is 😄

Reply:
The function is returning a slice. So we should make a new slice instead of updating the original one.

Reply:
I tend to reduce the cost of allocating new slide (since this one only be used to build string query) but you're right, to not change on the original slide is safer. Let's me change others `refineXXX` functions too.

Reply:
Address on https://github.com/pipe-cd/pipe/pull/1744/commits/7f501806e3cae28577273f3451b84b725ab9cbf6 🙏